status-checks/check.ts

```diff
@@ -0,0 +1,448 @@
+          );
+        }
+      }
+      const reqs = ["@dev", "@notice"];
+      for (let i3 = 0; i3 < reqs.length; i3++) {
+        if (!func.comments.some((comment) => comment.startsWith(reqs[i3]))) {
+          errors.push(
+            `no ${reqs[i3]} for function ${func.name} at ${connector.path}/main.sol:${func.firstLine}`
+          );
+        }
+      }
+    }
+    return errors;
+  } catch (error) {
+    return Promise.reject(error);
+  }
+};
+
+const checkPublicFuncs = async (connector: {
+  path: any;
+  allPublicFuncs?: any;
+}) => {
+  try {
+    const errors = [];
+    for (let i1 = 0; i1 < connector.allPublicFuncs.length; i1++) {
+      const { raw, firstLine, name } = connector.allPublicFuncs[i1];
+      if (!raw.includes("payable")) {
+        errors.push(
+          `public function ${name} is not payable at ${connector.path}/main.sol:${firstLine}`
+        );
+      }
+    }
+    return errors;
+  } catch (error) {
+    return Promise.reject(error);
+  }
+};
+
+const checkName = async (connector: { path: any; code?: any }) => {
+  try {
+    const strs = connector.code.split("\n");
+    let haveName = false;
+    for (let index = strs.length - 1; index > 0; index--) {
+      const str = strs[index];
+      if (
+        str.includes("string") &&
+        str.includes("public") &&
+        str.includes("name = ")
+      ) {
+        haveName = true;
+      }
+    }
+    return haveName
+      ? []
+      : [`name variable missing in ${connector.path}/main.sol`];
+  } catch (error) {
+    return Promise.reject(error);
+  }
+};
+
+const checkHeadComments = async (connector: { path: any; code?: any }) => {
+  try {
+    const errors = [];
+    const strs = connector.code.split("\n");
+    let haveTitle = false;
+    let haveDev = false;
+    for (let index = 0; index < strs.length; index++) {
+      if (!strs[index].includes("{")) {
+        if (strs[index].includes("@title")) haveTitle = true;
+        if (strs[index].includes("@dev")) haveDev = true;
+      } else {
+        break;
+      }
+    }
+    if (!haveTitle) errors.push(`@title missing in ${connector.path}/main.sol`);
+    if (!haveDev) errors.push(`@dev missing in ${connector.path}/main.sol`);
+    return errors;
+  } catch (error) {
+    return Promise.reject(error);
+  }
+};
+
+async function checkMain() {
+  try {
+    const connectorsRootsDirsDefault = ["mainnet", "polygon"].map(
+      (v) => `contracts/${v}/connectors`
+    );
+    const customPathArg = process.argv.find((a) => a.startsWith("connector="));
+    const connectorsRootsDirs = customPathArg
+      ? [customPathArg.slice(10)]
+      : connectorsRootsDirsDefault;
+    const errors = [];
+    const warnings = [];
+    const connectors = await getConnectorsList(connectorsRootsDirs);
+    for (let index = 0; index < connectors.length; index++) {
+      const { forbiddenErrors, code } = await checkForbidden(
+        connectors[index].path
+      );
+      connectors[index].code = code;
```

Comment:
@ishanjain28  this line is giving error - 
![image](https://user-images.githubusercontent.com/63545809/144734912-7d633944-267f-45d0-8c79-d4d01ebb490b.png)